amplify/backend/function/getMembersForColony/src/index.js

```diff
@@ -0,0 +1,202 @@
+      skillId,
+      constants.AddressZero,
+      rootHash,
+    );
+  } catch (error) {
+    // Not throwing anything, as its possible that the domain does not have
+    // any reputation, and we don't want to break the whole query
+  }
+
+  /*
+   * Validate Colony addresses
+   */
+  let checksummedAddress;
+  try {
+    checksummedAddress = utils.getAddress(colonyAddress);
+  } catch (error) {
+    throw new Error(
+      `Colony address "${colonyAddress}" is not valid (after checksum)`,
+    );
+  }
+
+  // get list of watchers for colony
+  let watchers = [];
+  if (domainId === ALL_DOMAIN_ID || domainId === ROOT_DOMAIN_ID) {
+    const { data, errors } = await graphqlRequest(
+      getWatchersInColony,
+      { id: checksummedAddress },
+      GRAPHQL_URI,
+      API_KEY,
+    );
+
+    if (errors || !data) {
+      const [error] = errors;
+      throw new Error(
+        error?.message || 'Could not fetch colony data from DynamoDB',
+      );
+    }
+
+    // Remove members that have reputation (i.e. contributors) to
+    // leave only watchers
+    watchers = data?.getColonyByAddress?.items[0]?.watchers.items.filter(
+      (item) =>
+        addressesWithReputation?.every(
+          (address) => address.toLowerCase() !== item.user.id.toLowerCase(),
+        ),
+    );
+  }
+
+  // get reputation for each address
+  const contributors = await Promise.all(
+    addressesWithReputation.map(async (address) => {
+      const { data } = await graphqlRequest(
+        getUser,
+        {
+          id: utils.getAddress(address),
+        },
+        GRAPHQL_URI,
+        API_KEY,
+      );
+      const user = data?.getUserByAddress?.items[0] || {};
+
+      try {
+        const userReputationForAllDomains =
+          await colonyClient.getReputationAcrossDomains(address, rootHash);
+
+        // Filter based on domainId if provided
+        const filteredReputationForAllDomains =
+          userReputationForAllDomains.filter(
+            (userReputation) =>
+              userReputation.reputationAmount &&
+              !userReputation.reputationAmount.isZero() &&
+              userReputation.domainId === domainId,
+          );
+
+        // Extract only the relevant data and
+        // transform the user reputation amount on each domain to percentage
+        const formattedUserReputations = filteredReputationForAllDomains.map(
+          (userReputation) => {
+            if (!userReputation?.reputationAmount) {
+              return {};
+            }
+
+            const reputationPercentage = calculatePercentageReputation(
+              userReputation.reputationAmount?.toString(),
+              totalColonyReputation.reputationAmount?.toString(),
+            );
+
+            return {
+              reputationPercentage,
+              reputationAmount: userReputation.reputationAmount?.toString(),
+            };
+          },
+        );
+
+        return {
+          user: user || {},
```

Comment:
The user object already defaults to an empty object, when instantiated: 
https://github.com/JoinColony/colonyCDapp/blob/b5466c2ce9f87ac44696d91dd5f34a2702d0d20b/amplify/backend/function/getMembersForColony/src/index.js#L124

Why do we need this fallback again ?